training_config/benchmark_iter.jsonnet

```diff
@@ -0,0 +1,145 @@
+local NUM_GPUS = 2;
+local NUM_THREADS = 10;
+
+local BASE_READER = {
+        "type": "simple_language_modeling",
+        "tokenizer": {
+          "type": "word",
+          "word_splitter": {
+	        // The 1 Billion Word Language Model Benchmark dataset is
+	        // pre-tokenized. (Also, if you're running against a untokenized
+	        // dataset be aware that there are serialization issues with Spacy.
+	        // These come into play in the multiprocess case.)
+            "type": "just_spaces"
+          }
+        },
+        "token_indexers": {
+          "tokens": {
+            "type": "single_id"
+          },
+          "token_characters": {
+            "type": "elmo_characters"
+          }
+        },
+        "max_sequence_length": 400,
+        "start_tokens": ["<S>"],
+        "end_tokens": ["</S>"]
+};
+
+local BASE_ITERATOR = {
+  "type": "bucket",
+  "max_instances_in_memory": 16384 * NUM_GPUS,
+  // Larger than we really desire for a batch. Since we set
+  // maximum_samples_per_batch below we will pack approximately that many
+  // samples in every batch.
+  "batch_size": 512 * NUM_GPUS,
+  "sorting_keys": [["source", "num_tokens"]],
+  "maximum_samples_per_batch": ["num_tokens", 2000]
```

Comment:
FYI this `maximum_samples_per_batch` makes looking at the perf here confusing, because within a single batch, it chops it up into potentially N batches and then yields them all really fast (because they have been tensorised already) so it means you get a stop-start type iteration, which makes it look like one of the queues is not being filled fast enough - when I think actually it's just the speed of yielding the chunked single batch.